sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamCalciteTable.java

```diff
@@ -38,32 +41,70 @@
 import org.apache.calcite.rex.RexNode;
 import org.apache.calcite.schema.ModifiableTable;
 import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.schema.Statistic;
+import org.apache.calcite.schema.Statistics;
 import org.apache.calcite.schema.TranslatableTable;
 
 /** Adapter from {@link BeamSqlTable} to a calcite Table. */
 public class BeamCalciteTable extends AbstractQueryableTable
     implements ModifiableTable, TranslatableTable {
   private final BeamSqlTable beamTable;
-  private final Map<String, String> pipelineOptions;
+  private final Map<String, String> pipelineOptionsMap;
+  private PipelineOptions pipelineOptions;
 
-  BeamCalciteTable(BeamSqlTable beamTable, Map<String, String> pipelineOptions) {
+  //  BeamCalciteTable(BeamSqlTable beamTable, Map<String, String> pipelineOptionsMap) {
+  //    this(beamTable,pipelineOptionsMap,null);
+  //  }
+
+  BeamCalciteTable(
+      BeamSqlTable beamTable,
+      Map<String, String> pipelineOptionsMap,
+      PipelineOptions pipelineOptions) {
     super(Object[].class);
     this.beamTable = beamTable;
+    this.pipelineOptionsMap = pipelineOptionsMap;
     this.pipelineOptions = pipelineOptions;
   }
 
   public static BeamCalciteTable of(BeamSqlTable table) {
-    return new BeamCalciteTable(table, ImmutableMap.of());
+    return new BeamCalciteTable(table, ImmutableMap.of(), null);
   }
 
   @Override
   public RelDataType getRowType(RelDataTypeFactory typeFactory) {
     return CalciteUtils.toCalciteRowType(this.beamTable.getSchema(), typeFactory);
   }
 
+  private PipelineOptions getPipelineOptions() {
```

Comment:
Is this the only change from #8822 ? Can you make it a separate commit?

Reply:
No. The pipelineoptions object was not passed to BeamCalciteTable. So the process in JDBC Driver that was converting pipeline options to map is also deleted. So JDBC Connection and JDBC Driver are also changed.